File: C++/KartikeyV/Trees/EvenOddTree.cpp
int solve(TreeNode* root){
    queue<TreeNode*>q;
    q.push(root);
    int prev = 0;
    bool evenLevel=true;

    while(!q.empty()){
        int sz = q.size();
        if(evenLevel) prev = 0;
        else prev = 10000000;
        for(int i = 0 ; i < sz ; i++){
            TreeNode* ele = q.front(); q.pop();
            int val = ele->val;
            if(evenLevel){
                if(val>prev and val%2==1)
                    prev = val;
                else 
                    return 0;
            }else{
                if(val<prev and val%2==0)
                    prev = val;
                else
                    return 0;
            }
            if(ele->left) q.push(ele->left);
            if(ele->right) q.push(ele->right);
        }
        evenLevel = !evenLevel;
    }
    return 1;
}

File: java/ankit_kanyal/devnest_problems/phase_2/EvenOddTree.java
// link- https://devsnest.in/algo-challenges/even-odd-tree?tab=question

static int solve(TreeNode root){
  Queue<TreeNode> que=new LinkedList<>();
    que.offer(root);
    boolean idxstatus=true;
    int level=-1;
    int prev=0;
        
//  true==odd && false==even
        
    while(que.size()>0){
        int size=que.size();
        level++;
        idxstatus=level%2==0;
            
        for(int i=0;i<size;i++){
            TreeNode node=que.poll();
            int value=node.val;
            if(idxstatus){
                if(value%2==0) return 0;
                if(value<=prev) return 0;
                prev=value;
            }else{
                if(value%2==1) return 0;
                if(value>=prev) return 0;
                prev=value;
            }
                
            if(node.left!=null)que.offer(node.left);
            if(node.right!=null)que.offer(node.right);
             
            }
        if(idxstatus){
            prev=10000000;
        }else{
            prev=0;
        }
    }
        
    return 1;
}